crates/doc/src/reduce/strategy.rs

```diff
@@ -301,6 +309,54 @@ impl Strategy {
         }
     }
 
+    fn json_schema_merge<'alloc, L: AsNode, R: AsNode>(
```

Comment:
nit: please place in a different file, similar to how the set reduction lives in `set.rs`.